libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -821,17 +825,47 @@ class MapboxNavigation @VisibleForTesting internal constructor(
                 RoutesExtra.ROUTES_UPDATE_REASON_CLEAN_UP,
                 initialLegIndex
             )
+            navigationSession.isPreview && directionsSession.routes == routes -> {
+                BasicSetRoutesInfo(RoutesExtra.ROUTES_UPDATE_REASON_NEW, initialLegIndex)
+            }
             routes.first() == directionsSession.routes.firstOrNull() ->
                 SetAlternativeRoutesInfo(initialLegIndex)
             else -> BasicSetRoutesInfo(RoutesExtra.ROUTES_UPDATE_REASON_NEW, initialLegIndex)
         }
+        navigationSession.isPreview = false
```

Comment:
Don't we have to do it inside `internalSetNavigationRoutes`? It can be called as a result of refresh or reroute. The routes will be set but the flag won't be unset.